jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/util/GenericErrorSummaryCountCell.java

```diff
@@ -64,32 +74,43 @@ public void render(Context context,
         } else {
             tooltipClasses.remove(ERROR_PRESENT_STYLE);
         }
-
-        if (viewPresenter.getViewErrorsActionCondition().test(value)) {
+        
+        if (viewPresenter.isUserAuthorizedForPerspective(EXECUTION_ERRORS) && errCount > 0) {
             tooltip.setTitle(Constants.INSTANCE.ErrorCountNumberView(errCount));
-            tooltip.setAttribute(PROCESS_INSTANCE_ATTRIBUTE,
-                                 Long.toString(value.getProcessInstanceId()));
+            tooltip.setAttribute((TASK_TYPE.equals(type)) ? TASK_INSTANCE_ATTRIBUTE : PROCESS_INSTANCE_ATTRIBUTE,
+                                 value.getId().toString());
             tooltipClasses.add(LINK_AVAILABLE_STYLE);
 
             initTooltipsAsync();
 
         } else {
             tooltip.removeAttribute("title");
-            tooltip.removeAttribute(PROCESS_INSTANCE_ATTRIBUTE);
+            tooltip.removeAttribute((TASK_TYPE.equals(type)) ? TASK_INSTANCE_ATTRIBUTE : PROCESS_INSTANCE_ATTRIBUTE);
             tooltipClasses.remove(LINK_AVAILABLE_STYLE);
         }
 
         sb.appendHtmlConstant(tooltip.getOuterHTML());
     }
 
-    public void openErrorView(final String pid) {
-        viewPresenter.openErrorView(pid);
+    public void openErrorView(final String id){
+        final PlaceRequest request = new DefaultPlaceRequest(EXECUTION_ERRORS);
+        
+        if(TASK_TYPE.equals(type)) {
+            request.addParameter(SEARCH_PARAMETER_TASK_ID,
+                                 id);
+        } else {
+            request.addParameter(SEARCH_PARAMETER_PROCESS_INSTANCE_ID,
+                                 id);
```

Comment:
@mcivantos-tribalyte I'm failing to see the benefits of the increased complexity here. Why are we checking the error type when it is clear form the DB query that the error type we are exclusively interested in is the task one:
```"errInfo.error_type=\'Task\'"```
This means that the error count associated with the specific task will only show the number of Task type errors. This is the information we want to show to task admins, right?
When we click on that number, the Execution errors view needs to match that number in order to maintain the consistency between the error count displayed in the admin task list and the number of errors listed in Execution errors view. To achieve this, the parameters being added to the request have to match the DB query in _DataSetDefsBootstrap_
Am I missing some use case scenario?

Reply:
@bsremac  Yes, you are right, the information we want to show in Task Admin view is the number of Task type error. So, this query (https://github.com/kiegroup/jbpm-wb/pull/965/files#diff-51ef7bbb4674fdc3238f8ec3494c30a6L116) is related to the Task Admin view where we are already filtering by "errInfo.error_type=\'Task\'", as you said.
When we click on that number we navigate to the Execution errors view. There, the query is (https://github.com/kiegroup/jbpm-wb/blob/1aea7ee22f5468f2bcaaef0936ad4f2041c99486/jbpm-wb-executor-service/jbpm-wb-executor-service-backend/src/main/java/org/jbpm/workbench/es/backend/server/DataSetDefsBootstrap.java#L92) and we were filtering by ProcessInstanceId when we navigate from Process List View and I added  request.addParameter(SEARCH_PARAMETER_TASK_ID, id); in order to filter by the taskId when you are clicking when we navigate from Task Admin View. 
So the use case scenarios that I've understood are:
- Navigate from Process List view, in that case we have to show all the errors related to this process.
- Navigate from Task Admin View: in that case we have to show only the errors related to this task.
I guess, you suggest that the correct behaviour is showing all the task errors related to one process instead of the errors related to the task you are selecting when clicking.
@cristianonicolai  what's the correct behaviour?

Reply:
@mcivantos-tribalyte No, I'm not suggesting any change in the DB query, with that being retrieving only task type errors for the particular task we are selecting. If that is what we want, that is fine. I just wanted to confirm that based on your latest changes, because a task can generate various types of errors (not only Type: Task) depending on the process design, task work item handlers etc.

Now going back to the actual implementation of the navigation to Execution Errors. What looks troubling in the introduced logic is that we are now deciding on the set of active filters in Execution Errors view based on some task type parameter we introduced just so we are able to differentiate where we are navigating from.
My suggestion is to either:
1. Split this method into 2 methods in which case one would handle only navigation from Process Instances View and the other only navigation from Task Admin View or
2.  If we want to stay on the track of having one universal method which will cover both navigation paths, introduce a clearer (unambiguous) criterion based on which the differentiation is being made. 
The current implementation assumes that a task can generate only task type error. This means that if a certain task has generated a task error, but also some other at the same time, when we navigate to Execution Errors view the list will show both of them and not only task type error (because both errors belong to the same task id).

Reply:
@bsremac Ok, I understand it. In the jira related to this PR, we only talk about Error_Type = 'Task' (https://issues.jboss.org/browse/JBPM-6174). So, I am going to implement the changes you suggest on the navigate to Execution Errors method in order to show only  Error_Type = 'Task'.

Reply:
@bsremac  Done

Reply:
@mcivantos-tribalyte in here, I believe this class should not dependend on the domain ( task or process ), you could simply delegate back to the presenter that would know which paramters to add into the place request. Same goes for the _TASK_INSTANCE_ATTRIBUTE_ and _PROCESS_INSTANCE_ATTRIBUTE_ that could be generic as _data-jbpm-id_.